akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -458,6 +549,53 @@ class ShardRegion(
     }
   }
 
+  def replyToRegionStateQuery(ref: ActorRef): Unit = {
+    askAllShards[Shard.CurrentShardState](Shard.GetCurrentShardState).map { shardStates ⇒
+      CurrentShardRegionState(shardStates.map {
+        case (shardId, state) ⇒ ShardRegion.ShardState(shardId, state.entityIds)
+      }.toSet)
+    }.recover {
+      case x: AskTimeoutException ⇒ CurrentShardRegionState(Set.empty)
+    }.pipeTo(ref)
+  }
+
+  def replyToRegionStatsQuery(ref: ActorRef): Unit = {
+    askAllShards[Shard.ShardStats](Shard.GetShardStats).map { shardStats ⇒
+      ShardRegionStats(shardStats.map {
+        case (shardId, stats) ⇒ (shardId, stats.entityCount)
+      }.toMap)
+    }.recover {
+      case x: AskTimeoutException ⇒ ShardRegionStats(Map.empty)
+    }.pipeTo(ref)
+  }
+
+  def askAllShards[T: ClassTag](msg: Any): Future[Seq[(ShardId, T)]] = {
+    implicit val timeout: Timeout = 3.seconds
+    Future.sequence(shards.toSeq.map {
+      case (shardId, ref) ⇒ (ref ? msg).mapTo[T].map(t ⇒ (shardId, t))
+    })
+  }
+
+  def replyToClusterShardingStats(waitMax: FiniteDuration, ref: ActorRef): Unit = {
+    implicit val timeout: Timeout = waitMax
+    val regionsAndSelf = regions.keySet + self
```

Comment:
A region does not know about all other regions unless it has asked the coordinator for the `GetShardHome`, which will be done when the shard is first used in this specific region. I think the stats must be aggregated via the coordinator to be correct.